Error running the ra-tls-mbedtls example

### Description of the problem

While running the command 'ARCH_LIBDIR=/lib64 make app' in RHEL 8.6 server, the following error was observed :-

```
gramine-manifest \
        -Dlog_level=error \
        -Darch_libdir=/lib/x86_64-redhat-linux \
        -Dra_type=none \
        -Dra_client_spid= \
        -Dra_client_linkable=0 \
        server.manifest.template > server.manifest
cc src/server.c -O2 -fPIE -I./mbedtls/include -pie -ldl -Wl,--enable-new-dtags -Wl,-rpath,/usr/lib64 -lmbedcrypto_gramine -lmbedtls_gramine -lmbedx509_gramine  -o server
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/libmbedtls_gramine.a(ssl_tls.o): In function `mbedtls_ssl_conf_dh_param_ctx':
(.text+0x2055): undefined reference to `mbedtls_dhm_get_value'
(.text+0x2069): undefined reference to `mbedtls_dhm_get_value'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/libmbedtls_gramine.a(ssl_tls.o): In function `mbedtls_ssl_handshake_step':
(.text+0x280c): undefined reference to `mbedtls_ssl_states_str'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/libmbedtls_gramine.a(ssl_tls.o): In function `mbedtls_ssl_handshake_free':
(.text+0x29c8): undefined reference to `mbedtls_dhm_free'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/libmbedtls_gramine.a(ssl_tls.o): In function `ssl_handshake_init':
(.text+0x3f43): undefined reference to `mbedtls_dhm_init'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/libmbedtls_gramine.a(ssl_tls.o): In function `mbedtls_ssl_parse_sig_alg_ext':
(.text+0x4ad0): undefined reference to `mbedtls_ssl_sig_alg_to_str'
(.text+0x4bca): undefined reference to `mbedtls_ssl_sig_alg_to_str'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/libmbedtls_gramine.a(ssl_tls.o): In function `mbedtls_ssl_psk_derive_premaster':
(.text+0x5226): undefined reference to `mbedtls_dhm_calc_secret'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/libmbedtls_gramine.a(ssl_tls.o): In function `mbedtls_ssl_write_sig_alg_ext':
(.text+0x6b42): undefined reference to `mbedtls_ssl_sig_alg_to_str'
(.text+0x6bc9): undefined reference to `mbedtls_ssl_sig_alg_to_str'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/libmbedtls_gramine.a(ssl_tls12_server.o): In function `ssl_parse_client_dh_public':
(.text+0x43): undefined reference to `mbedtls_dhm_read_public'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/libmbedtls_gramine.a(ssl_tls12_server.o): In function `mbedtls_ssl_handshake_server_step':
(.text+0x323f): undefined reference to `mbedtls_dhm_set_group'
(.text+0x3274): undefined reference to `mbedtls_dhm_get_len'
(.text+0x3292): undefined reference to `mbedtls_dhm_make_params'
(.text+0x3964): undefined reference to `mbedtls_dhm_calc_secret'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/libmbedtls_gramine.a(ssl_tls12_client.o): In function `mbedtls_ssl_handshake_client_step':
(.text+0x1469): undefined reference to `mbedtls_dhm_get_len'
(.text+0x14b4): undefined reference to `mbedtls_dhm_get_len'
(.text+0x14cf): undefined reference to `mbedtls_dhm_make_public'
(.text+0x1559): undefined reference to `mbedtls_dhm_calc_secret'
(.text+0x2833): undefined reference to `mbedtls_dhm_read_params'
(.text+0x284b): undefined reference to `mbedtls_dhm_get_bitlen'
(.text+0x2df3): undefined reference to `mbedtls_dhm_get_len'
(.text+0x2e57): undefined reference to `mbedtls_dhm_get_len'
(.text+0x2e73): undefined reference to `mbedtls_dhm_make_public'
collect2: error: ld returned 1 exit status
make: *** [Makefile:80: server] Error 1
```
Thanks

### Steps to reproduce

_No response_

### Expected results

_No response_

### Actual results

_No response_

### Gramine commit hash

e18bc05b17fd704b259cb0401f928dc4ec5199a6

--- a/subprojects/packagefiles/mbedtls/meson.build
+++ b/subprojects/packagefiles/mbedtls/meson.build
@@ -46,9 +46,11 @@ pkgconfig.generate(
     libraries: [
         '-L${libdir}',
         '-Wl,-rpath,${libdir}',
+        '-Wl,--start-group',
         '-lmbedcrypto_gramine',
         '-lmbedtls_gramine',
         '-lmbedx509_gramine',
+        '-Wl,--end-group',
     ],
 )
 

